internal/kubernetes/client.go

```diff
@@ -136,6 +137,81 @@ func (c *client) Apply(u *unstructured.Unstructured) (Metadata, error) {
 	return metadata, nil
 }
 
+// Replace a given manifest.
+func (c *client) Replace(u *unstructured.Unstructured) (Metadata, error) {
+	metadata := Metadata{}
+	gvk := u.GroupVersionKind()
+
+	restMapping, err := c.mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+	if err != nil {
+		return metadata, err
+	}
+
+	gvr := restMapping.Resource
+	gv := gvk.GroupVersion()
+	c.config.GroupVersion = &gv
+
+	restClient, err := newRestClient(*c.config, gv)
+	if err != nil {
+		return metadata, err
+	}
+
+	helper := resource.NewHelper(restClient, restMapping)
+
+	info := &resource.Info{
+		Client:          restClient,
+		Mapping:         restMapping,
+		Namespace:       u.GetNamespace(),
+		Name:            u.GetName(),
+		Source:          "",
+		Object:          u,
+		ResourceVersion: restMapping.Resource.Version,
+	}
+
+	// If annotation kubectl.kubernetes.io/last-applied-configuration exists, then update it.
+	err = util.CreateOrUpdateAnnotation(false, info.Object, unstructured.UnstructuredJSONScheme)
+	if err != nil {
+		return metadata, err
+	}
```

Comment:
I know this is in the `kubectl` source code as well, but I thought the replace command was to get away from setting this annotation... can you help me understand what's going on here?

Reply:
We did some testing with `kubectl` and if this annotation is present in the inbound manifest, then `kubectl` kept the annotation, but corrected the value, on the `replace`.  There's really no reason someone would be passing in  the `last-applied-configuration` annotation into Spinnaker, but we decided to keep the same behavior as `kubectl`.

If we wanted to do some big-brain stuff... 

We could calculate the size of the value, and if it was too big for kubernetes, then automatically delete it regardless if the replace strategy annotation was set or not.  This would make it easier when pulling in CRDs for helm charts where you may not be given the option to add an annotation.